org.opent2t.onboarding.hue/js/thingOnboarding.js

```diff
@@ -0,0 +1,175 @@
+                    //extract nonce code from header
+                    var digestHeader = err.response.headers['www-authenticate'];
+                    var nonce = digestHeader.substr(digestHeader.indexOf('nonce=\"') + 7, 32);
+
+                    //Compute digest header response
+                    var disgestHeaderContent = 'username=\"' + authInfo[0].client_id + '\", ';
+                    disgestHeaderContent += 'realm=\"oauth2_client@api.meethue.com\", ';
+                    disgestHeaderContent += 'nonce=\"' + nonce + '\", ';
+                    disgestHeaderContent += 'uri=\"/oauth2/token\", ';
+                    var HASH1 = crypto.createHash('md5').update(authInfo[0].client_id + ':oauth2_client@api.meethue.com:' + authInfo[0].client_secret).digest('hex');
+                    var HASH2 = crypto.createHash('md5').update('POST:/oauth2/token').digest('hex');
+                    var authHeaderResponse = crypto.createHash('md5').update(HASH1 + ':' + nonce + ':' + HASH2).digest('hex');
+                    disgestHeaderContent += 'response=\"' + authHeaderResponse + '\"';
+
+                    options.headers = {
+                        'Accept': 'application/json',
+                        'Authorization': 'Digest ' + disgestHeaderContent,
+                    };
+
+                    return request(options).then(function (body) {
+                        var tokenInfo = JSON.parse(body); // This includes refresh token, scope etc..
+                        var getOptions = {
+                            url: 'https://api.meethue.com/v2/bridges',
+                            method: "GET",
+                            headers: {
+                                'Authorization': 'Bearer ' + tokenInfo.access_token
+                            },
+                        };
+
+                        console.log('Getting bridge ID associated to the account...');
+
+                        return request(getOptions)
+                            .then(function (body) {
+
+                                //Get bridgeId
+                                var bridgeIds = JSON.parse(body);
+                                if (bridgeIds.length > 0) {
+
+                                    console.log('Success.');
+
+                                    var putOptions = {
+                                        url: 'https://api.meethue.com/v2/bridges/' + bridgeIds[0].id + '/0/config',
+                                        method: "PUT",
+                                        headers: {
+                                            'Authorization': 'Bearer ' + tokenInfo.access_token,
+                                            'Content-Type': 'application/json'
+                                        },
+                                        json: { 'linkbutton': true }
+                                    };
+
+                                    console.log('Getting whitelist ID from bridge ' + bridgeIds[0].id + '...');
```

Comment:
For Hue's remote API, we would need to obtain the bridge ID and whitelist ID to be able to anything (i.e. query for device or hub details, update device property) through the Hue bridge. I put this part here because I do not if there is a better place to put it.